zig/src/context.zig

```diff
@@ -0,0 +1,33 @@
+const std = @import("std");
+const c = @import("ffi.zig");
+const utils = @import("utils.zig");
+const toCstr = utils.toCstr;
+
+pub const Context = struct {
+    mutex: std.Thread.Mutex,
+    ctx: *c.ExtismContext,
+
+    // We have to use this until ziglang/zig#2647 is resolved.
+    error_info: ?[]const u8,
```

Comment:
I read through the issue, thank you for adding this note. I _think_ I understand, but to clarify, this `error_info` field could eventually move into an error set union such as: 

```zig
const ExtismError = union(error) {
  Context: []const u8,
  Plugin: []const u8,
  // ... etc
}
```

ref: https://github.com/ziglang/zig/issues/2647

Reply:
I want to have a union like that too but AFAIK, having a field for reporting errors is the best option we have right now

Reply:
Oh, I only meant to ask that you confirm my basic understanding of the proposal you mentioned :) 

I see this is the way we must implement it for now. Thanks!